src/utils/miscellaneous.ts

```diff
@@ -88,7 +88,50 @@ export async function silentPromise<T>(promise: Promise<T>, customMessage?: stri
   try {
     return await promise
   } catch (e) {
-    console.error(customMessage || 'Failed to fetch promise', e)
+    logDebug(customMessage || 'Failed to fetch promise', e.message)
     return
   }
 }
+
+interface RetryFnParams<T> {
+  fn: (...fnParams: unknown[]) => Promise<T>
+  fnParams?: unknown[]
```

Comment:
This kind of typing allows for incorrect inputs like
```ts
retry({
  fn: async (a: number, b: number) => a + b,
  fnParams: ['', true],
})
```
You'll have to use type inference to make it work. Kinda like [here](https://github.com/gnosis/dex-price-estimator/blob/a4c2e34b6f94378df937916cc65c96040d25a7c3/src/utilities.ts#L57L58)